crates/core/c8y_api/src/http_proxy.rs

```diff
@@ -107,7 +107,11 @@ impl C8yEndPoint {
 
         let (_, host) = url_host.split_once('.').unwrap_or((url_host, ""));
         let (_, c8y_host) = tenant_uri.split_once('.').unwrap();
-        (host == c8y_host).then_some(url)
+
+        // The configured `c8y.http` setting may have a port value specified,
+        // but the incoming URL is less likely to have any port specified.
+        // Hence just matching the host prefix.
+        (c8y_host.starts_with(host)).then_some(url)
```

Comment:
question: Wouldn't this condition also be satisfied by an empty string, which is probably not what we want?

`host` will be empty if the URL doesn't contain any `.` characters, e.g. `http://localhost`, and the function will return `Some(_)`, when it probably should return `None`. If I'm correct, then we should also add such case to the unit tests below.

Reply:
Good catch!

This is why the system test "tedge-agent should terminate on SIGINT while downloading file" is failing!

The url `http://localhost/speedlimit/10MB` is erroneously taken as an url to the tenant and rewritten `http://127.0.0.1:8001/c8y/speedlimit/10MB`, preventing the download to start.